File: MPI_Example/template_MPI.cpp
#include<iostream>
#include<mpi.h>

int main(int argc,char* argv[])
{

	/*     MPI start    */
	int nproc,rank=0;
	MPI_Init(&argc,&argv);
	MPI_Comm_size(MPI_COMM_WORLD,&nproc);
	MPI_Comm_rank(MPI_COMM_WORLD,&rank);
	
	/*     System setting    */
	int n_ensemble;	
		
	//Ensemble loop - Parallelized!
	for ( int i = rank; i < n_ensemble; i += nproc )
	{
	
		/*     Calculate here    */

	
	}
	
	//MPI finish
	MPI_Finalize();
	return 0;
}

File: MPI_Example/template_MPI.py
#System setting
n_max=10000

#MPI start when mpi4py is imported
from mpi4py import MPI

comm = MPI.COMM_WORLD
nproc = comm.Get_size()
rank = comm.Get_rank()

#Ensemble loop - parallized
for i in range(rank,nmax,nproc):
	#Calculate here

#Finish MPI
MPI.COMM_WORLD.Barrier()
MPI.Finalize()
